spec/grover/middleware_spec.rb

```diff
@@ -383,11 +383,14 @@ def request_env
             with(
               'Grover McGroveryface',
               display_url: 'http://www.example.org/test',
-              cookies: [{ domain: 'www.example.org', name: 'key', value: 'value' }]
+              cookies: [
+                { domain: 'www.example.org', name: 'key', value: 'value' },
+                { domain: 'www.example.org', name: 'escaped', value: '%26%3D%3D'}
```

Comment:
seems like a reasonable test, however i'd suggest it would also be good to have a test that actually calls to Grover/Puppeteer with some encoded cookies, have the page render the contents then check the result contains the expected encoded values. Hopefully that way there will be less risk of future regressions. This would probably live in the processor_spec

Reply:
Agreed. I shall make it so.

Reply:
Added that spec, but all it really does it prove that you can pass a cookies, escaped or not to Puppeteer, the real failure on this was not realising that the cookie parser had escaped it. My bad.